app/controllers/application_controller.rb

```diff
@@ -186,4 +186,12 @@ def redirect_old_urls
       redirect_to @node.path, status: :moved_permanently
     end
   end
+
+  def signed_in?
+    !!current_user
+  end
+
+  def current_user=(user)
```

Comment:
I am having trouble here. Trying to make it work. Any help appreciated.

Reply:
If this error is solved, it means we are done with login via a provider

Reply:
OH, can you paste in the error you're having? Perhaps highlight exactly where this is done by the existing login system?

Reply:
yeah, I marked it just now